mm2src/coins/lp_coins.rs

```diff
@@ -600,11 +600,12 @@ impl TradePreimageError {
             GenerateTxError::OutputValueLessThanDust { value, dust } => {
                 if is_upper_bound {
                     // If the preimage value is [`TradePreimageValue::UpperBound`], then we had to pass the account balance as the output value.
-                    let error = format!(
-                        "Output value {} (equal to the account balance) less than dust {}. Probably, dust is not set or outdated",
-                        value, dust
-                    );
-                    TradePreimageError::InternalError(error)
+                    let required = big_decimal_from_sat_unsigned(dust, decimals);
+                    TradePreimageError::NotSufficientBalance {
+                        coin,
+                        available: big_decimal_from_sat_unsigned(value, decimals),
+                        required,
+                    }
```

Comment:
This change was made after a discussion with @sergeyboyko0791

Reply:
As we discussed before, if the value was upper-bound, then we tried to generate a transaction with `max=true`. And if the max value is even lower than `dust`, it means that we either don't have sufficient balance (it's possible if the value is zero), or the dust was increased (internal error), or Electrum server has invalid state or something else (internal error).
I only realized it now, so how about checking if the value is zero or not @Milerius?
```rust
if is_upper_bound {
  if value == 0 {
    // NotSufficientBalance
  } else {
    // InternalError
  }
}
```